app/src/main/java/com/android/calendar/AllInOneActivity.java

```diff
@@ -1408,6 +1415,61 @@ public void handleEvent(EventInfo event) {
                 }
             }
             displayTime = event.startTime.toMillis();
+        } else if (event.eventType == EventType.VIEW_TASK) {
+
+            // If in Agenda view and "show_event_details_with_agenda" is "true",
+            // do not create the event info fragment here, it will be created by the Agenda
+            // fragment
+
+            if (mCurrentView == ViewType.AGENDA && mShowEventDetailsWithAgenda) {
+                if (event.startTime != null && event.endTime != null) {
+                    // Event is all day , adjust the goto time to local time
+                    if (event.isAllDay()) {
+                        Utils.convertAlldayUtcToLocal(event.startTime, event.startTime.toMillis(), mTimeZone);
+                        Utils.convertAlldayUtcToLocal(event.endTime, event.endTime.toMillis(), mTimeZone);
+                    }
+                    mController.sendEvent(this, EventType.GO_TO, event.startTime, event.endTime, event.selectedTime, event.id, ViewType.AGENDA, CalendarController.EXTRA_GOTO_TIME, null, null);
+                } else if (event.selectedTime != null) {
+                    mController.sendEvent(this, EventType.GO_TO, event.selectedTime, event.selectedTime, event.id, ViewType.AGENDA);
+                }
+            } else {
+                // TODO Fix the temp hack below: && mCurrentView !=
```

Comment:
do you plan to fix this before finishing the PR?

Reply:
it is not my TODO. it already exists